lib/Data/Aeson/Combinators/Decode.hs

```diff
@@ -303,6 +304,17 @@ instance MonadFail Decoder where
   {-# INLINE fail #-}
 
 
+-- | Conversely, an Aeson's 'FromJSON' instance can be implemented by using 'Decoder' combinators.
+--
+-- > newtype People = People [Person]
+-- >
+-- > instance FromJSON People where
+-- >     parseJSON = fromDecoder $ Decode.list personDecoder
+fromDecoder :: Decoder a -> Value -> Parser a
+fromDecoder (Decoder f) = f
+{-# INLINE fromDecoder #-}
```

Comment:
alternatively you can have also just change definition of type to:

```hs
newtype Decoder a =
Decoder { fromDecoder :: (Value -> Parser a) }
```

to get this function "for free". But there is no harm in having separate documentation from type for this so I'm OK with it.

Reply:
Yeah! In fact I wrote it this way to follow the same style as in the `Encode` module, which has a `run` function that unwraps an `Encoder`.